File: Array/objc_based_array_sol/factorial.m
#import <Foundation/Foundation.h>

NSInteger factorialOfNumber(NSInteger number) {
    if (number == 0 || number == 1) {
        return 1;
    } else {
        return number * factorialOfNumber(number - 1);
    }
}

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        NSInteger result = factorialOfNumber(5);
        NSLog(@"%ld", result);
    }
    return 0;
}

File: Array/swift_based_array_sol/factorial.swift
import Foundation

func factorialOfNumber(withNumber number: Int) -> Int {
    if number == 0 || number == 1 {
        return 1
    } else {
        return number * factorialOfNumber(withNumber: number - 1)
    }
}

let result = factorialOfNumber(withNumber: 5)
print(result)
